### DISCUSSION TITLE
Scrolling is Laggy

### DISCUSSION THREAD

**Discussion by priyaan99** [helix-editor/helix#10240] [General]
I'm using helix for 3 months. 
Moving cursor up and down feels laggy.

**Comment by David-Else** [comment]
I have not noticed it in particular, but I think this could improve things: https://github.com/helix-editor/helix/issues/731

**Comment by wongjiahau** [comment]
Are you running Helix in Zellij?

**Comment by priyaan99** [comment]
I'm running helix in Alacritty and default Pop Os terminal emulator.

**Comment by mek-yt** [comment]
I moved out from Alacritty because of visual tearing and the laggy feeling in helix. 
On my compositor Sway, Alacritty and Foot have low performance whereas Kitty and Wezterm have no tearing and lag.

**Comment by kchibisov** [comment]
>On my compositor Sway, Alacritty and Foot have low performance whereas Kitty and Wezterm have no tearing and lag.

tearing doesn't mean low performance in terminals, usually such terminals are faster because they can draw more frames per second and both alacritty and foot has one of the fastest vte parsers/rendering performance.

To avoid tearing the #731 CSI(or DCS) escape sequence should be used, because it's a way to do atom updates to the terminal. foot supports CSI sequence for example, and alacritty right now only handles the original DCS one.
